spec/tests/run_agent.rb

```diff
@@ -1,8 +1,7 @@
 require 'spec_helper'
 
 
-describe command('/var/rudder/cfengine-community/bin/cf-agent -KI ' + $params['PARAMS'] + ' | grep -v "error: Warning promised, need to create file"') do
+describe command('/opt/rudder/bin/rudder agent ' + $params['PARAMS']) do
   its(:exit_status) { should eq 0 }
-  its(:stdout) { should_not match /error:/ }
```

Comment:
Because we now handle the exit status correctly. We could still match "error:", but on stderr instead, to match cases where CFEngine outputs an error and the promise is not failed.